packages/toolpad-app/src/toolpad/Apps/index.tsx

```diff
@@ -199,23 +199,55 @@ function CreateAppDialog({ onClose, open, ...props }: CreateAppDialogProps) {
 
         sendAppCreatedEvent(createdApp.name, appTemplateId);
       } catch (rawError) {
-        if (config.recaptchaV2SiteKey && !hasShownRecaptchaCheckbox && captchaTargetRef.current) {
-          const error = errorFrom(rawError);
-          if (error.code === ERR_VALIDATE_CAPTCHA_FAILED) {
-            // Show captcha checkbox
-            const widgetId = grecaptcha.render(captchaTargetRef.current.id, {
-              sitekey: config.recaptchaV2SiteKey,
-            });
-            latestRecaptchaWidgetIdRef.current = widgetId;
-          }
+        const error = errorFrom(rawError);
+        if (config.recaptchaV2SiteKey && error.code === ERR_VALIDATE_CAPTCHA_FAILED) {
+          // Show captcha checkbox
+          setRequestRecaptchaV2(true);
+        } else {
+          throw error;
```

Comment:
Seems good to throw here, I think we were suppressing the error before but I wasn't sure if it was intentional.